src/dispatch/decorators.py

```diff
@@ -45,7 +48,8 @@ def wrapper(*args, **kwargs):
             print(traceback.format_exc())
             sentry_sdk.capture_exception(e)
         finally:
-            db_session.close()
+            if background:
+                kwargs["db_session"].close()
```

Comment:
There are two flows, as a web request (inline) that has an established db_session for the web request and and background task (spawns a new thread and does not have a db_session). If we don't check to see we are calling it in the background (by checking that db_session doesn't exist) we run into issues where the session gets closed by this function, even though functions call it may still need to use the session.